Symfony/CS/Error/Error.php

```diff
@@ -0,0 +1,73 @@
+<?php
+
+/*
+ * This file is part of the PHP CS utility.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace Symfony\CS\Error;
+
+/**
+ * An abstraction for errors that can occur before and during fixing.
+ *
+ * @author Andreas Möller <am@localheinz.com>
+ *
+ * @internal
+ */
+class Error
+{
+    /**
+     * Error which has occurred in linting phase, before applying any fixers.
+     */
+    const TYPE_INVALID = 1;
+
+    /**
+     * Error which has occurred during fixing phase.
+     */
+    const TYPE_EXCEPTION = 2;
+
+    /**
+     * Error which has occurred in linting phase, after applying any fixers.
+     */
+    const TYPE_LINT = 3;
+
+    /**
+     * @var string
+     */
+    private $type;
+
+    /**
+     * @var string
+     */
+    private $filePath;
+
+    /**
+     * @param string $type
+     * @param string $filePath
+     */
+    public function __construct($type, $filePath)
+    {
```

Comment:
Maybe make the `filePath` optional (default 'null') ? It make the error object more re-usable for example for configurations errors and such?